webcompat/webhooks/model.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""WebCompat Issue Model for webhooks."""
+
+from dataclasses import dataclass, field, InitVar
+from typing import Any, Dict, List
+
+from webcompat import app
+from webcompat.webhooks.helpers import extract_metadata
+from webcompat.webhooks.helpers import get_issue_labels
+from webcompat.webhooks.helpers import make_request
+from webcompat.webhooks.helpers import prepare_rejected_issue
+
+PUBLIC_REPO = app.config['ISSUES_REPO_URI']
+PRIVATE_REPO = app.config['PRIVATE_REPO_URI']
+
+
+@dataclass
+class WebHookIssue:
+    """WebCompat Issue Model for WebHook consumption"""
+    payload: InitVar[Dict[str, Any]]
+    action: str = field(init=False)
+    body: str = field(init=False)
+    domain: str = field(init=False)
+    number: int = field(init=False)
+    public_url: str = field(init=False)
+    repository_url: str = field(init=False)
+    state: str = field(init=False)
+    title: str = field(init=False)
+    original_labels: List[str] = field(init=False)
+    milestone: str = ''
+    milestoned_with: str = ''
+
+    def close_private_issue(self):
+        """Mark the private issue as closed."""
+        path = f'repos/{PRIVATE_REPO}/{self.number}'
+        self.state = 'closed'
+        make_request('patch', path, {'state': self.state})
```

Comment:
> if the request fails, then we have the issue with a closed state when it is still open remotely

Ah, I did not consider this because GitHub *never* fails. 😈